docs/content/node-operation/full-observer-node.mdx

```diff
@@ -1,38 +1,221 @@
+The [Flow cli](https://developers.flow.com/tools/flow-cli/install) can be used to generate this key.
+
+e.g.
+```
+flow keys generate
+Private Key  1fc3d962ec947bcc9d8dfae6b1e2b54dbdd208bbca0c7af094d5d52c7991257f
+Public Key   774e3812f031d57f767a0afa79bd2cfad8a99b3d20b039f6583f82a5dbc5a208be70aa616ddfa1c7676ccac0b4e4e7da78ebdac6ab0d47cbae1efec95f66a7a5
+```
+
+Store the private key in a file under the bootstrap subdirectory
+e.g.
+```
+$ cat flow_observer/bootstrap/network.key
+1fc3d962ec947bcc9d8dfae6b1e2b54dbdd208bbca0c7af094d5d52c7991257f
+```
+
+### Step 3 Download the root-protocol-state-snapshot.json file for the current spork
+
+The `root-protocol-state-snapshot.json` is generated for each [spork](https://developers.flow.com/nodes/node-operation/spork) and contains the genesis data for that spork.
+It is published and made available after each spork. The download location is specified [here](https://github.com/onflow/flow/blob/master/sporks.json) under [rootProtocolStateSnapshot](https://github.com/onflow/flow/blob/master/sporks.json#L16) and can be downloaded as follows,
+
+For mainnet-18
+```
+wget -P bootstrap/public-root-information https://storage.googleapis.com/flow-genesis-bootstrap/mainnet-18-execution/public-root-information/root-protocol-state-snapshot.json
+wget -P bootstrap/public-root-information https://storage.googleapis.com/flow-genesis-bootstrap/mainnet-18-execution/public-root-information/root-protocol-state-snapshot.json.asc
+```
+
+For testnet-37
+```
+wget -P bootstrap/public-root-information https://storage.googleapis.com/flow-genesis-bootstrap/testnet-37/public-root-information/root-protocol-state-snapshot.json
+wget -P bootstrap/public-root-information https://storage.googleapis.com/flow-genesis-bootstrap/testnet-37/public-root-information/root-protocol-state-snapshot.json.asc
+```
+
+#### Verify the PGP signature
+
+Add the `flow-signer@onflow.org` public key
+```
+gpg --keyserver keys.openpgp.org --search-keys flow-signer@onflow.org
+```
+
+Verify the root-snapshot file
+```
+gpg --verify bootstrap/public-root-information/root-protocol-state-snapshot.json.asc
+```
+
+```
+gpg: assuming signed data in 'bootstrap/public-root-information/root-protocol-state-snapshot.json'
+gpg: Signature made Wed Sep 15 11:34:33 2021 PDT
+gpg:                using ECDSA key 40CD95717AC463E61EE3B285B718CA310EDB542F
+gpg: Good signature from "Flow Team (Flow Full Observer node snapshot verification master key) <flow-signer@onflow.org>" [unknown]
+gpg: WARNING: This key is not certified with a trusted signature!
+gpg:          There is no indication that the signature belongs to the owner.
+Primary key fingerprint: 7D23 8D1A E6D3 2A71 8ECD  8611 CB52 64F7 FD4C DD27
+     Subkey fingerprint: 40CD 9571 7AC4 63E6 1EE3  B285 B718 CA31 0EDB 542F
+```
+
+Alternately, if you don't care about the blocks before the current block, you can request the current root-snapshot file via the flow-cli.
```

Comment:
We should note here that the genesis block information MUST be from a trusted source as it's the foundation for all block verification. So if they choose this method, they must use an AN they trust.

Reply:
The flow-cli actually supports using the secure connection. I switched the command to use that so that it is guaranteed that the snapshot file the user gets is the real one.